### PR TITLE
Drew.new up: Fixes runtime problems, adds ranked choice voting

### PR DESCRIPTION
**Pull request by drewstone**

### CODE REVIEW

File: modules/edge-voting/src/voting.rs [hunk 1]
```diff
@@ -139,35 +138,45 @@ decl_module! {
...
 		/// There are currently no cryptoeconomic incentives for revealing commited votes.
-		pub fn reveal(origin, vote_id: u64, vote: VoteOutcome, secret: Option<VoteOutcome>) -> Result {
+		pub fn reveal(origin, vote_id: u64, vote: Vec<VoteOutcome>, secret: Option<VoteOutcome>) -> Result {
 			let _sender = ensure_signed(origin)?;
 			let mut record = <VoteRecords<T>>::get(vote_id).ok_or("Vote record does not exist")?;
 			ensure!(record.data.stage == VoteStage::Voting, "Vote is not in voting stage");
-			// Check vote is for a valid outcome
-			ensure!(record.outcomes.iter().any(|o| o == &vote), "Vote outcome is not valid");
+			// Check vote is for valid outcomes
+			for i in 0..vote.len() {
```
Comment (jnaviask) [inline modules/edge-voting/src/voting.rs]: What happens if I vote for multiple items in a non-ranked choice vote?

File: setup.sh [hunk 2]
```diff
@@ -26,6 +26,7 @@ fi
 rustup update nightly
 rustup target add wasm32-unknown-unknown --toolchain nightly
 rustup update stable
+rustup default nightly
```
Comment (jnaviask) [inline setup.sh]: Is nightly necessary for the current release?

Comment (drewstone) [reply setup.sh]: I can remove, may have added that accidentally

File: modules/edge-voting/src/voting.rs (hunk 1 above)
Comment (drewstone) [reply modules/edge-voting/src/voting.rs]: Addressed

File: README.md [hunk 3]
```diff
@@ -97,8 +97,7 @@ To ensure you followed the steps correctly, check https://telemetry.polkadot.io/
 
 * [Aura](https://github.com/paritytech/substrate/tree/master/srml/aura)
 * [Balances](https://github.com/paritytech/substrate/tree/master/srml/balances)
-* [Consensus](https://github.com/paritytech/substrate/tree/master/srml/consensus)
-* [Contract](https://github.com/paritytech/substrate/tree/master/srml/contract)
+* [Contracts](https://github.com/paritytech/substrate/tree/master/srml/contracts)
 * [Council](https://github.com/paritytech/substrate/tree/master/srml/council)
```
Comment (jnaviask) [inline README.md]: We need to update the README for the new modules.

Comment (drewstone) [reply README.md]: Right right

**Pull request merged**